helios-testing/src/main/java/com/spotify/helios/testing/ExistingHeliosDeployment.java

```diff
@@ -0,0 +1,237 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.testing;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+
+import com.spotify.helios.client.HeliosClient;
+import com.spotify.helios.common.descriptors.Job;
+import com.spotify.helios.common.descriptors.JobId;
+import com.spotify.helios.common.descriptors.JobStatus;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.lang.System.getProperty;
+import static java.util.Arrays.asList;
+
+/**
+ * This class represents an existing Helios cluster.
+ */
+public class ExistingHeliosDeployment implements HeliosDeployment {
+
+  private static final Logger log = LoggerFactory.getLogger(ExistingHeliosDeployment.class);
+
+  private static final String DEFAULT_USER = getProperty("user.name");
+  private static final String DEFAULT_PREFIX_DIRECTORY = "/tmp/helios-temp-jobs";
+
+  private final HeliosClient heliosClient;
+  private final JobPrefixFile jobPrefixFile;
+  private final Undeployer undeployer;
+
+  private ExistingHeliosDeployment(final Builder builder) {
+    this.heliosClient = checkNotNull(builder.heliosClient, "heliosClient");
+    final Path prefixDirectory = Paths.get(firstNonNull(builder.prefixDirectory,
+                                                        DEFAULT_PREFIX_DIRECTORY));
+
+    try {
+      removeOldJobs(asList(firstNonNull(prefixDirectory.toFile().listFiles(), new File[0])));
+      if (isNullOrEmpty(builder.jobPrefix)) {
+        this.jobPrefixFile = JobPrefixFile.create(prefixDirectory);
+      } else {
+        this.jobPrefixFile = JobPrefixFile.create(builder.jobPrefix, prefixDirectory);
+      }
+    } catch (IOException | ExecutionException | InterruptedException e) {
+      throw Throwables.propagate(e);
+    }
+
+    this.undeployer = firstNonNull(builder.undeployer, new DefaultUndeployer(heliosClient));
+  }
+
+  @Override
+  public HeliosClient client() {
+    return heliosClient;
+  }
+
+  @Override
+  public Set<URI> uris() {
+    return Collections.emptySet();
+  }
+
+  @Override
+  public void cleanup() {
+    jobPrefixFile.delete();
+  }
+
+  @Override
+  public void close() {
+    try {
+      heliosClient.close();
+    } catch (IOException e) {
+      log.warn("HeliosClient did not close cleanly: {}", e);
```

Comment:
if you pass a throwable as the last argument to a log method, logback (or slf4j, not sure which) interprets it as you wanting to log the stacktrace, so this will look like

```
WARN HeliosClient did not close cleanly: {}
java.util.IOException: blah
  at foo.bar.blah():123
  at ...
```

if you just want to log the message and not the stacktrace do `log.warn("HeliosClient did not close cleanly: {}", e.toString())`